### ISSUE TITLE
update function for settings 

### ISSUE THREAD

**Issue by Newan** [akveo/ng2-smart-table#337]
It is only possible to update the complete object like:
`this.settings = {....}`

It is not possible to update an single parameter like:

`this.settings.noDataMessage = "No User found for ..."`

**Comment by Ezeon** [comment]
You are right. You can only change the whole object. You can of course get around this by having a separate object for your settings which you can change and then assigning a copy of this object to this.settings when you do a change. That's at least what I do in my application.

Lets say you create a object mySettings:
mySettings = {} // remember to add all the properties you need
You can then do some changes to it:
e.g. mySettings.data.actions.add = false;
You then need to assign a copy of it to this.settings:
this.settings = Object.assign({}, mySettings)

**Comment by Newan** [comment]
thanks for this workaround

**Comment by ppenan** [comment]
I followed this, and it didn't work

**Comment by Romsan01** [comment]
Wrap the "smart table" in ngIf and wait for the data to load :)

**Comment by Newan** [comment]
Hello,

I have the same problem again. Normal it works, but updating it into an arrow function like API call or timeout does not work:

Example:

```
ngOnInit() {
    setTimeout(() => {
      console.log("Not work!!");
      this.changeSettings("not work!");
    }, 2000);
    
      console.log("Work!!");
      this.changeSettings();
  }

  changeSettings(name: string) {
    this.mySettings.noDataMessage =  name;
    this.settings = Object.assign({}, this.mySettings)
  }
```

**Comment by 128keaton** [comment]
@Newan I am in the same boat!

**Comment by vanessasoutoc** [comment]
Great. Work for me @Ezeon.
this.settings = Object.assign ({}, mySettings)
